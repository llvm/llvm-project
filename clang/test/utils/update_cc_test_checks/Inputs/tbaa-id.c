// RUN: %clang_cc1 -triple=x86_64-unknown-linux-gnu -O1 -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s

void foo(unsigned **ptr) {
  *ptr = 0;
}
