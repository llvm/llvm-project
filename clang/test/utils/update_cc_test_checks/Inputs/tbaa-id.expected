// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
// RUN: %clang_cc1 -triple=x86_64-unknown-linux-gnu -O1 -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s

// CHECK-LABEL: @foo(
// CHECK-NEXT:  entry:
// CHECK-NEXT:    [[PTR_ADDR:%.*]] = alloca i32**, align 8
// CHECK-NEXT:    store i32** [[PTR:%.*]], i32*** [[PTR_ADDR]], align 8, !tbaa [[TBAA2:![0-9]+]]
// CHECK-NEXT:    [[TMP0:%.*]] = load i32**, i32*** [[PTR_ADDR]], align 8, !tbaa [[TBAA2]]
// CHECK-NEXT:    store i32* null, i32** [[TMP0]], align 8, !tbaa [[TBAA2]]
// CHECK-NEXT:    ret void
//
void foo(unsigned **ptr) {
  *ptr = 0;
}
